analyzers/src/SonarAnalyzer.Common/Common/AnalyzerLanguage.cs

```diff
@@ -156,5 +159,13 @@ language switch
                 ".VB" => VisualBasic,
                 _ => None
             };
+
+        public static AnalyzerLanguage FromName(string name) =>
+            name switch
+            {
+                LanguageNames.CSharp => CSharp,
+                LanguageNames.VisualBasic => VisualBasic,
+                _ => None
```

Comment:
Given the weird class design, it makes sense to return `None` for now. I think I'll be able to remove `None` and `Both` in the future and then it will make sense to throw here.